python/cudf/cudf/core/index.py

```diff
@@ -1362,6 +1294,15 @@ def from_pandas(cls, index, nan_as_null=None):
         ind.name = index.name
         return ind
 
+    def _reduce(
+        # self, op, axis=None, level=None, numeric_only=None, **kwargs,
+        self,
+        op,
+        *args,
+        **kwargs,
+    ):
+        raise AttributeError(f"{type(self)} object has no attribute {op}.")
```

Comment:
Is there a compelling reason to do this other than matching the behavior of Pandas? Why shouldn't we expose the reductions on the Index classes if we can for free?

Reply:
I don't have a strong opinion either way here, so I went with [@shwina's suggestion](https://github.com/rapidsai/cudf/pull/8944#issuecomment-892633947). I'll let him respond rather than trying to put words in his mouth.

Reply:
One less thing for us to worry about maintaining? Internally, we shouldn't ever call public methods like `Series.max()` or `Index.max()`. Externally, users aren't expecting this behavior anyway.

Reply:
Just because Pandas doesn't support an operation doesn't mean users aren't expecting the behavior, but it looks like many of these reductions are supported by Pandas, see `methods` here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.html

Reply:
`pandas` seems to support a somewhat random subset of reductions. I'm not sure how to interpret which ones, but I agree that if it's trivial to enable (I don't think it requires any additional maintenance since `Series` and `Index` can share this logic) we may as well allow them. I'll reenable for now, at the moment it's actually easier to keep it than to actively disable for `Index` types.

Reply:
That's fine with me